File: haskell/02_01_fib.hs
fib :: Integer -> Integer
fib 0 = 0
fib 1 = 1
fib x = fib (x - 1) + fib (x - 2)

main = do
  print (fib 0 == 0)
  print (fib 1 == 1)
  print (fib 5 == 5)
  print (fib 10 == 55)

File: rust/02_01_fib.rs
pub fn fib(n: i32) -> i32 {
    if n < 2 {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn main() {
    let actual = fib(10);
    let expected = 55;

    assert_eq!(actual, expected);
}
